[Tensorpipe Agent] Dist Autograd Tests for Tensorpipe Agent

osalpekar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #38447 [Tensorpipe Agent] Enabling tests with OSS CI
* #38446 [Tensorpipe Agent] Dist Optimizer Tests for Tensorpipe Agent
* **#38445 [Tensorpipe Agent] Dist Autograd Tests for Tensorpipe Agent**
* #38444 [Tensorpipe Agent] RPC, RRef tests for Tensorpipe Agent

This PR enables the Distributed Autograd tests for the Tensorpipe Agent. A decorator is used to skip all tests that are currently failing due to functionality lacking in the Tensorpipe RPC Agent (primarily timeouts and error handling).

Differential Revision: [D21560098](https://our.internmc.facebook.com/intern/diff/D21560098/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21560098/)!

osalpekar committed:
[Tensorpipe Agent] Dist Autograd Tests for Tensorpipe Agent

This PR enables the Distributed Autograd tests for the Tensorpipe Agent. A decorator is used to skip all tests that are currently failing due to functionality lacking in the Tensorpipe RPC Agent (primarily timeouts and error handling).

Differential Revision: [D21560098](https://our.internmc.facebook.com/intern/diff/D21560098/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21560098/)!

[ghstack-poisoned]

lw commented on torch/testing/_internal/distributed/rpc/dist_autograd_test.py:
```diff
@@ -938,6 +950,8 @@ def _test_backward_rref(self, callee, rref_owner):
         local_grads = None
         t1 = torch.ones((3, 3), requires_grad=True)
         t2 = torch.zeros((3, 3), requires_grad=True)
+        print("inside function")
+        print(dist_utils.TEST_CONFIG.rpc_backend_name)
```
I guess leftover debug prints?

lw commented on torch/testing/_internal/distributed/rpc/dist_autograd_test.py:
```diff
@@ -26,7 +26,9 @@
-
+from torch.testing._internal.distributed.rpc.tensorpipe_rpc_agent_test_fixture import (
+    TensorPipeRpcAgentTestFixture,
+)
 
 # Right now we test up to 3-layer nested rpc calls.
```
Does the style guide for PyTorch mandate two blank lines after the imports? If so, you're removing one of them

osalpekar committed:
Update on "[Tensorpipe Agent] Dist Autograd Tests for Tensorpipe Agent"


This PR enables the Distributed Autograd tests for the Tensorpipe Agent. A decorator is used to skip all tests that are currently failing due to functionality lacking in the Tensorpipe RPC Agent (primarily timeouts and error handling).

Differential Revision: [D21560098](https://our.internmc.facebook.com/intern/diff/D21560098/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21560098/)!

[ghstack-poisoned]

osalpekar committed:
Update on "[Tensorpipe Agent] Dist Autograd Tests for Tensorpipe Agent"


This PR enables the Distributed Autograd tests for the Tensorpipe Agent. A decorator is used to skip all tests that are currently failing due to functionality lacking in the Tensorpipe RPC Agent (primarily timeouts and error handling).

Differential Revision: [D21560098](https://our.internmc.facebook.com/intern/diff/D21560098/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21560098/)!

[ghstack-poisoned]

osalpekar committed:
Update on "[Tensorpipe Agent] Dist Autograd Tests for Tensorpipe Agent"


This PR enables the Distributed Autograd tests for the Tensorpipe Agent. A decorator is used to skip all tests that are currently failing due to functionality lacking in the Tensorpipe RPC Agent (primarily timeouts and error handling).

Differential Revision: [D21560098](https://our.internmc.facebook.com/intern/diff/D21560098/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21560098/)!

[ghstack-poisoned]

osalpekar committed:
Update on "[Tensorpipe Agent] Dist Autograd Tests for Tensorpipe Agent"


This PR enables the Distributed Autograd tests for the Tensorpipe Agent. A decorator is used to skip all tests that are currently failing due to functionality lacking in the Tensorpipe RPC Agent (primarily timeouts and error handling).

Differential Revision: [D21560098](https://our.internmc.facebook.com/intern/diff/D21560098/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21560098/)!

[ghstack-poisoned]

osalpekar committed:
Update on "[Tensorpipe Agent] Dist Autograd Tests for Tensorpipe Agent"


This PR enables the Distributed Autograd tests for the Tensorpipe Agent. A decorator is used to skip all tests that are currently failing due to functionality lacking in the Tensorpipe RPC Agent (primarily timeouts and error handling).

Differential Revision: [D21560098](https://our.internmc.facebook.com/intern/diff/D21560098/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21560098/)!

[ghstack-poisoned]

osalpekar committed:
Update on "[Tensorpipe Agent] Dist Autograd Tests for Tensorpipe Agent"


This PR enables the Distributed Autograd tests for the Tensorpipe Agent. A decorator is used to skip all tests that are currently failing due to functionality lacking in the Tensorpipe RPC Agent (primarily timeouts and error handling).

Differential Revision: [D21560098](https://our.internmc.facebook.com/intern/diff/D21560098/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21560098/)!

[ghstack-poisoned]

osalpekar committed:
Update on "[Tensorpipe Agent] Dist Autograd Tests for Tensorpipe Agent"


This PR enables the Distributed Autograd tests for the Tensorpipe Agent. A decorator is used to skip all tests that are currently failing due to functionality lacking in the Tensorpipe RPC Agent (primarily timeouts and error handling).

Differential Revision: [D21560098](https://our.internmc.facebook.com/intern/diff/D21560098/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21560098/)!

[ghstack-poisoned]

osalpekar committed:
Update on "[Tensorpipe Agent] Dist Autograd Tests for Tensorpipe Agent"


This PR enables the Distributed Autograd tests for the Tensorpipe Agent. A decorator is used to skip all tests that are currently failing due to functionality lacking in the Tensorpipe RPC Agent (primarily timeouts and error handling).

Differential Revision: [D21560098](https://our.internmc.facebook.com/intern/diff/D21560098/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D21560098/)!

[ghstack-poisoned]
